### PR TITLE
Documentation improvements

### PR DESCRIPTION
**Pull request by RickWinter**

### CODE REVIEW

File: CMakeSettings.json [hunk 1]
```diff
@@ -1,40 +1,45 @@
...
-      "generator": "Ninja",
-      "configurationType": "Debug",
-      "inheritEnvironments": [ "msvc_x64_x64" ],
-      "buildRoot": "${projectDir}\\out\\build\\${name}",
-      "installRoot": "${projectDir}\\out\\install\\${name}",
-      "cmakeCommandArgs": "",
-      "buildCommandArgs": "-v",
-      "ctestCommandArgs": "",
-      "variables": [
+    "configurations": [
```
Comment (ahsonkhan) [inline CMakeSettings.json:2]: Revert the changes to this file?

Comment (RickWinter) [reply CMakeSettings.json]: VS loves to reformat that file, frustrating

File: sdk/inc/azure/storage/az_storage_blobs.h [hunk 2]
```diff
@@ -92,16 +90,14 @@ AZ_NODISCARD az_result az_storage_blobs_blob_client_init(
...
+ * @brief Allows customization of the upload operation.
  */
 typedef struct
 {
   az_context* context; ///< Operation context.
   struct
   {
-    /// Currently, this is unused, but needed as a placeholder since we can't have an empty struct.
-    az_span unused;
+    bool unused;
```
Comment (RickWinter) [inline sdk/inc/azure/storage/az_storage_blobs.h:100]:
```suggestion
    /// Currently, this is unused, but needed as a placeholder since we can't have an empty struct.
    bool unused;
```

File: sdk/inc/azure/storage/az_storage_blobs.h [hunk 3]
```diff
@@ -128,7 +125,7 @@ AZ_NODISCARD AZ_INLINE az_storage_blobs_blob_upload_options
 az_storage_blobs_blob_upload_options_default()
 {
   return (az_storage_blobs_blob_upload_options){ .context = &az_context_application,
-                                                 ._internal = { .unused = AZ_SPAN_NULL } };
+                                                 ._internal = { .unused = true } };
```
Comment (ahsonkhan) [inline sdk/inc/azure/storage/az_storage_blobs.h:128]: `unused` should be set to false as that is what the "default" value would be if we did:
```C
az_storage_blobs_blob_upload_options options = { 0 };
```

Comment (ahsonkhan) [reply sdk/inc/azure/storage/az_storage_blobs.h:128]:
```suggestion
                                                 ._internal = { .unused = false } };
```
